### ISSUE TITLE
403 Forbidden error

### ISSUE THREAD

**Issue by DerDangla** [bridgecrewio/prisma-cloud-jetbrains-ide#150]
Hi,

I installed the Prisma Cloud plugin in IntelliJ and set up the keys and API. However, I encountered the following error while scanning a simple TF file on my desktop.

Please note that the token I’m using has only read-only privileges in Prisma, which might be contributing to the issue.

`Error while scanning framework terraform, original error message - Error while scanning terraform, exit code - 0, error reason - ERROR] Get fixes request for file /C[--colon--]Users/[user]/Desktop/prisma-test/providers.tf failed with response code 403: Forbidden - skipping fixes for this file . Please check the log file in C:\Users\[user]\AppData\Local\Temp\checkov2839043246234428848\error-terraform--debug-output4596695228787613285.txt. Prisma Cloud result can be found in C:\Users\[user]\AppData\Local\Temp\checkov2839043246234428848\error-terraform--checkov-result15396937086850659782.json. To report: open a issue at https://github.com/bridgecrewio/prisma-cloud-jetbrains-ide/issues`

**Comment by ChananM** [comment]
Hi @DerDangla,
Can you please run the checkov scan directly from the terminal? If it also fails, then we will be able to clasify whether this is a Checkov/API issue or an issue with the IDE plugin.
To see the exact scan command, you can view the plugin logs by clicking the button next to the scan button on the plugin view in the IDE. Please note the bc API key is censored in the log and you will need to fill it in the terminal

**Comment by ChananM** [comment]
@DerDangla 

1. That makes me wonder how you got the `Error while scanning framework terraform` message. If you don't have any log mentioning a scan, how could it fail? I also expected to find the error you first mentioned in the logs.
2. As for read only API key, it should still work with Checkov. I can also see from your log you failed to call the analytics API with 403. Please provide any `--repo-id` for the command and run again to see it it reproduces. The flag should be in the format of `<org>/<repo>`, but it doesn't matter much. You can put the value `extension/jetbrains`.

**Comment by DerDangla** [comment]
My apologies, I might have mixed up two different issues:
1. The first error was encountered when I was installing checkov with docker. I tried to run the command shown from the logs and here is the result

command (I updated the bc-api-key):
`docker run --rm -a stdout -a stderr --env BC_SOURCE=jetbrains --env BC_SOURCE_VERSION=1.0.23-203.241 --env PRISMA_API_URL=**-**-**-** --volume C:\Users\[user]\AppData\Local\Temp\checkov1428568393330157666\cdk-checkov-result6600102405031314292.json:/C[--colon--]Users/[user]/AppData/Local/Temp/checkov1428568393330157666/cdk-checkov-result6600102405031314292.json --volume C:/Users/[user]/Desktop/prisma-test:/C[--colon--]Users/[user]/Desktop/prisma-test bridgecrew/checkov -d C[--colon--]Users/edangla/Desktop/prisma-test --skip-path node_modules -s --bc-api-key **-**-**-** --repo-id jetbrains/extension --quiet -o cli -o json --output-file-path console,C[--colon--]Users/[user]/AppData/Local/Temp/checkov1428568393330157666/cdk-checkov-result6600102405031314292.json --prisma-api-url https://api.ca.prismacloud.io --framework cdk`

result:
`docker: Error response from daemon: invalid mode: /Users/[user]/AppData/Local/Temp/checkov1428568393330157666/cdk-checkov-result6600102405031314292.json.`

2. The second error was encountered when I'm installing checkov using python/pip and I this is the python and pip version I have
```
pip 24.3.1 from C:\Users\[user]\AppData\Local\Programs\Python\Python313\Lib\site-packages\pip (python 3.13)
Python 3.13.0
```

**Comment by DerDangla** [comment]
This appears to be an issue on my installed python version `3.13.0`. The checkov installation using python was resolved when I downgrade to python 3.12.7 (which is a more stable version as of today).

For the `403: Forbidden - skipping fixes` error, it appears to be related to api key... My api key has  an`ACCOUNT GROUP READ ONLY` role and when I use the same api key in VS Code, the `fix` button is not appearing in the `Prisma Issue` tab. Not sure if this makes sense but looking at the documentation, it says `no` access to `Fix and Submit Changes to repositories`. Not sure if this makes sense because I am scanning code from my local computer --> https://docs.prismacloud.io/en/enterprise-edition/content-collections/administration/prisma-cloud-admin-permissions#:~:text=Fix%20and%20Submit%20Changes%20to%20repositories

**Comment by ChananM** [comment]
Hi @DerDangla 
I'm not sure what the error from Docker means, but I can say that we are dropping support for docker installation soon, so it's redundant to try to investigate it. As for the Python version, you are correct, and it says so on the Checkov GitHub page: https://github.com/bridgecrewio/checkov?tab=readme-ov-file#python-version-support. Version 3.13 is not supported by Checkov. In a future release the IDE plugin will show an appropriate message to address this.
 `ACCOUNT GROUP READ ONLY` permission should work with the plugin API endpoints, but I can't attest to the integration with Checkov. If you keep getting those errors even when running the scan from the terminal, please contact Prisma Cloud support to contact the relevant team, as this is no longer an IDE plugin issue.

**Comment by DerDangla** [comment]
You're right; I completely missed that point about the Python version. It mentions compatibility only up to 3.11, but I found it works on 3.12.7 as well—it would be helpful to capture that in the documentation.

After further investigation, I discovered that read-only access cannot retrieve the fix, leading to the error. The Access Key requires one of the following permissions: Developer, AppSec Admin, or System Administrator. https://docs.prismacloud.io/en/enterprise-edition/content-collections/application-security/ides/connect-vscode#:~:text=To%20integrate%20the%20plugin%2C%20Developer%2C%20AppSec%20Admin%20or%20System%20Administrator%20roles%20are%20required.
